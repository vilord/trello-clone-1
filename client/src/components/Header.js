import React from 'react';
import PropTypes from 'prop-types';
import { Route, Link } from 'react-router-dom';

import { Icon, Image } from 'semantic-ui-react';
import Search from '../containers/Search';
import ButtonIcon from '../components/ButtonIcon';

import './Header.css';

const Header = ({
  avatar,
  showBoardsExplorer,
  showCreateMenu,
  showNotifications,
  showUserMenu,
}) => (
  <div className="Header">
    <div className="Header-left">
      <Route
        path="/b"
        render={() => (
          <Link to="/">
            <ButtonIcon icon="mail forward" flipped />
          </Link>
        )}
      />
      <ButtonIcon icon="trello" label="Boards" onClick={showBoardsExplorer} />
      <Search />
    </div>
    <div className="Header-title">
      <Link to="/">
        <Icon name="trello" size="large" />
        <h2>Trello</h2>
      </Link>
    </div>
    <div className="Header-right">
      <ButtonIcon icon="plus" onClick={showCreateMenu} />
      <ButtonIcon icon="bell" onClick={showNotifications} />
      <Image src={avatar} avatar onClick={showUserMenu} />
    </div>
  </div>
);

Header.propTypes = {
  avatar: PropTypes.string,
  showBoardsExplorer: PropTypes.func.isRequired,
  showCreateMenu: PropTypes.func.isRequired,
  showNotifications: PropTypes.func.isRequired,
  showUserMenu: PropTypes.func.isRequired,
};

export default Header;
