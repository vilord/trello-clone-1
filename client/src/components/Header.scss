.Header {
  position: absolute;
  top: 0;
  left: 0;

  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: space-between;

  width: 100vw;
  height: 40px;

  color: #84b3d2;
  background-color: #026aa7;
}

.Header-left {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  width: 40%;
  margin-left: 5px;;

  a {
    line-height: 2.8;
  }
}

.Header-title {
  display: flex;
  align-items: center;
  justify-content: center;

  width: 20%;
  padding: 5px;

  a {
    text-decoration: none;

    color: inherit;

    i {
      line-height: .6 !important;
    }
  }

  h2 {
    font-family: 'Damion', handwriting;

    display: inline;

    margin: 0;
  }

  &:hover {
    color: rgb(207, 225, 237);
  }
}

.Header-right {
  display: flex;
  justify-content: flex-end;

  width: 40%;
  margin-right: 5px;

  img {
    width: 30px !important;
    height: 30px !important;
    margin-left: 4px;

    cursor: pointer;
  }
}
