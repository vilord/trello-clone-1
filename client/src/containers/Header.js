import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, Link } from 'react-router-dom';

import { Icon, Image } from 'semantic-ui-react';
import Search from './Search';
import ButtonIcon from '../components/ButtonIcon';

import './Header.css';

export class Header extends Component {
  render() {
    return (
      <div className="Header">
        <div className="Header-left">
          <Route
            path="/"
            render={() => (
              <Link to="/">
                <ButtonIcon icon="mail forward" flipped />
              </Link>
            )}
          />
          <ButtonIcon icon="trello" label="Boards" />
          <Search />
        </div>
        <div className="Header-title">
          <Link to="/">
            <Icon name="trello" size="large" />
            <h2>Trello</h2>
          </Link>
        </div>
        <div className="Header-right">
          <ButtonIcon icon="plus" />
          <ButtonIcon icon="bell" />
          <Image src={this.props.avatar} avatar />
        </div>
      </div>
    );
  }
}

Header.defaultProps = {
  avatar: PropTypes.string,
};

Header.propTypes = {};

const mapStateToProps = state => ({
  avatar: state.user.avatar,
});

const mapDispatchToProps = dispatch => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
