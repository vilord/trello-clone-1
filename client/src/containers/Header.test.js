import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';

describe('Header Component', () => {
  beforeAll(() => {
    console.error = error => {
      throw new Error(error);
    };
  });

  let component;
  beforeEach(() => {
    component = shallow(<Header />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  xit('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
