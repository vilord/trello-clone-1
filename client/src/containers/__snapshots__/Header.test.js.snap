// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<div
  className="Header"
>
  <div
    className="Header-left"
  >
    <Button
      as="button"
      icon={true}
    >
      <Icon
        as="i"
        name="arrow left"
      />
    </Button>
    <Button
      as="button"
      icon={true}
      labelPosition="left"
    >
      <Icon
        as="i"
        name="trello"
      />
      Boards
    </Button>
    <div
      className="Header-search"
    >
      <Search
        fluid={true}
        icon="search"
        input="text"
        loading={false}
        minCharacters={1}
        noResultsMessage="No results found."
        showNoResults={true}
      />
    </div>
  </div>
  <div
    className="Header-title"
  >
    <Icon
      as="i"
      name="trello"
      size="large"
    />
    <h2>
      Trello
    </h2>
  </div>
  <div
    className="Header-right"
  >
    <Button
      as="button"
      icon={true}
    >
      <Icon
        as="i"
        name="plus"
      />
    </Button>
    <Button
      as="button"
      icon={true}
    >
      <Icon
        as="i"
        name="bell"
      />
    </Button>
    <Image
      as="img"
      avatar={true}
      ui={true}
    />
  </div>
</div>
`;
